composer.json

```diff
@@ -21,22 +21,33 @@
     }
   ],
   "require": {
-    "php": "^7.3 || ^8.0",
-    "ext-json": ">=1.3.7",
+    "php": "^8.1",
     "ext-curl": "*",
-    "ezimuel/ringphp": "^1.2.2",
+    "ext-json": ">=1.3.7",
+    "php-http/async-client-implementation": "^1.0",
```

Comment:
should we really require an async-client implementation when we might be using only the sync API ?

Maybe we should even go further than my previous comment and split the async and sync clients in separate packages, allowing them to have their own dependencies.

Reply:
It's a good idea, as long as it's not too much work. Generally, I am thinking introducing PSR interfaces is a key milestone that we need to hit so I'd want to make sure there's not crazy feature creep.

Reply:
Well, reading my comment on https://github.com/opensearch-project/opensearch-php/pull/219#discussion_r1715828607 again, the existing client is sync-only, so the MVP is actually to keep this client sync-only for now. If there is a need for an async PHP client, it can be added later